### PR TITLE
feat: add docs for News launches

### PR DESCRIPTION
**Pull request by scissorsneedfoodtoo**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

To do: add info about downloading the `dayjs` file for the new language as part of the PR to the CDN repo.

### CODE REVIEW

File: docs/how-to-enable-new-languages.md [hunk 1]
```diff
@@ -281,3 +281,101 @@ Once these are in place, you should be able to run `npm run develop` to view you
...
+article1title: 'Linux ln Example'
+article1link: 'https://www.freecodecamp.org/news/linux-ln-how-to-create-a-symbolic-link-in-linux-example-bash-command'
+article2title: 'JS document.ready()'
+article2link: 'https://www.freecodecamp.org/news/javascript-document-ready-jquery-example/'
+article3title: ...
+article3link: ...
+  ...
+```
+
+Open a PR to the CDN repo with these changes and tag `@freeCodeCamp/dev-team` for review.
```
Comment (naomi-lgbt) [inline docs/how-to-enable-new-languages.md:311]:
```suggestion
Open a PR to the CDN repo with these changes for review.
```

We don't need to tag - the CODEOWNERS auto-requests reviews in a round robin style.

File: docs/how-to-enable-new-languages.md [hunk 2]
```diff
@@ -281,3 +281,101 @@ Once these are in place, you should be able to run `npm run develop` to view you
...
+  arabic: 'news-ar',
+  bengali: 'news-bn',
+  chinese: 'news-zh',
+  english: 'news',
+  dothraki: 'news-do'
+};
+```
+
+> [!NOTE]
+> Note that the values above are in alphabetic order. Also, Dothraki does not actually have an ISO 639-1 code, so `do` is just used as an example.
```
Comment (naomi-lgbt) [inline docs/how-to-enable-new-languages.md:353]:
```suggestion
const localeCodes = {
  arabic: 'ar',
  bengali: 'bn',
  chinese: 'zh',
  english: 'en',
  dothraki: 'mis'
};

const algoliaIndices = {
  arabic: 'news-ar',
  bengali: 'news-bn',
  chinese: 'news-zh',
  english: 'news',
  dothraki: 'news-mis'
};
```

> [!NOTE]
> Note that the values above are in alphabetic order. Also, Dothraki does not actually have an ISO 639-1 code, so `mis` is just used as the uncoded language value.
```

Nitpicking, but ISO 369-2 has an actual code for non-coded languages.

File: docs/how-to-enable-new-languages.md [hunk 3]
```diff
@@ -281,3 +281,101 @@ Once these are in place, you should be able to run `npm run develop` to view you
...
+
+### Add the i18next JSON Files for the New Language
+
+Next, go to the `config/i18n/locales` directory, create a new folder, and give it the name of the new language you're adding. For example, if you're launching Dothraki News, create a new folder named `dothraki`.
+
+Then go to the `english` directory, copy the `links.json` and `meta-tags.json` files, and paste them into your new folder.
```
Comment (naomi-lgbt) [inline docs/how-to-enable-new-languages.md:359]:
```suggestion
Then go to the `english` directory, copy the JSON files, and paste them into your new folder.
```

As mentioned on the Indonesian news PR.

File: docs/how-to-enable-new-languages.md [hunk 4]
```diff
@@ -281,3 +281,101 @@ Once these are in place, you should be able to run `npm run develop` to view you
...
+
+```json
+{
+  "redirects": []
+}
+```
+
+> [!NOTE]
+> You do not need to create a `translation.json` file. This file will be downloaded from Crowdin before the new language launches.
+
```
Comment (naomi-lgbt) [inline docs/how-to-enable-new-languages.md:371]:
```suggestion

```

File: docs/how-to-enable-new-languages.md [hunk 5]
```diff
@@ -281,3 +281,101 @@ Once these are in place, you should be able to run `npm run develop` to view you
...
+article2title: 'JS document.ready()'
+article2link: 'https://www.freecodecamp.org/news/javascript-document-ready-jquery-example/'
+article3title: ...
+article3link: ...
+  ...
+```
+
+Open a PR to the CDN repo with these changes and tag `@freeCodeCamp/dev-team` for review.
+
+## Prep the News Repo for the New Language
```
Comment (naomi-lgbt) [inline docs/how-to-enable-new-languages.md:313]:
```suggestion
## Prep the News Repo for the New Language

> [!WARN]
> Pull requests to the news repo *must* come from the same repo. You should not work off of a fork for this step.
```

As an excuse to break up the two headers next to each other, but also this is important to note.

**Review by naomi-lgbt — DISMISSED** [review]

File: docs/how-to-enable-new-languages.md (hunk 5 above)
Comment (scissorsneedfoodtoo) [reply docs/how-to-enable-new-languages.md:313]: Awesome, this looks great. And it's a much better idea to add this warning up front like you did here.

File: docs/how-to-enable-new-languages.md (hunk 3 above)
Comment (scissorsneedfoodtoo) [reply docs/how-to-enable-new-languages.md]: Sure, this looks good, and should be easier to follow.

I'll just make sure to update the following section and say that the contents of the English `serve.json` file should be replaced by the object with the empty `redirects` array below.

File: docs/how-to-enable-new-languages.md (hunk 2 above)
Comment (scissorsneedfoodtoo) [reply docs/how-to-enable-new-languages.md]: Nice, I didn't realize that. TIL!

File: docs/how-to-enable-new-languages.md [hunk 6]
```diff
@@ -281,3 +281,99 @@ Once these are in place, you should be able to run `npm run develop` to view you
...
+  dothraki: 'mis'
+};
+
+const algoliaIndices = {
+  arabic: 'news-ar',
+  bengali: 'news-bn',
+  chinese: 'news-zh',
+  english: 'news',
+  dothraki: 'news-mis'
+};
```
Comment (naomi-lgbt) [inline docs/how-to-enable-new-languages.md:355]:
````suggestion
};
```
````

My fault - lost this in one of the suggestions

Comment (scissorsneedfoodtoo) [reply docs/how-to-enable-new-languages.md:386]: No worries, I was missing some backticks myself. It should be taken care of now.

**Comment by scissorsneedfoodtoo** [comment]
Thanks again for your review @naomi-lgbt, and for all of your helpful suggestions. This should be ready for another round of reviews.

@RafaelDavisH, could you also take a look at this when you get a chance? I was able to add some instructions on how to add a dayjs file for the new language in the CDN repo section.

File: docs/how-to-enable-new-languages.md [hunk 7]
```diff
@@ -281,3 +281,130 @@ Once these are in place, you should be able to run `npm run develop` to view you
...
+
+### Modify the Main Config File
+
+Clone the News repo and create a new branch.
+
+Open the `config/index.js` file to add the new language and configure the necessary values. There are a few objects and arrays to modify:
+
+- `locales`: This array contains the active and upcoming News languages. These are the values that are used in the `.env` file to choose the Ghost instance and UI to use for each build. Add the text name of the new language in lowercase to this array.
+- `localeCodes`: This object is a map of ISO codes for each language, and is used to configure i18next before building the UI. To add a new language, use the lowercase language name as the _key_ and the ISO 639-1 language code as the _value_.
+- `algoliaIndices`: This object is a map of Algolia indicies for each language. To add a new language, use the lowercase language name as the _key_, and `news-` followed by the lowercase ISO 639-1 language code as the _value_.
```
Comment (Sembauke) [inline docs/how-to-enable-new-languages.md:362]:
```suggestion
- `algoliaIndices`: This object is a map of Algolia indices for each language. To add a new language, use the lowercase language name as the _key_, and `news-` followed by the lowercase ISO 639-1 language code as the _value_.
```

**Review by Sembauke — DISMISSED** [review]

**Review by raisedadead — DISMISSED** [review]

**Review by Sembauke — DISMISSED** [review]

File: docs/how-to-enable-new-languages.md [hunk 8]
```diff
@@ -281,3 +281,130 @@ Once these are in place, you should be able to run `npm run develop` to view you
...
+article2title: 'JS document.ready()'
+article2link: 'https://www.freecodecamp.org/news/javascript-document-ready-jquery-example/'
+article3title: ...
+article3link: ...
+  ...
+```
+
+### Add a dayjs file for the New Language
+
+In the [`build/news-assets/dayjs/1.10.4/locale`](https://github.com/freeCodeCamp/cdn/tree/main/build/news-assets/dayjs/1.10.4/locale) directory, create a new file and name it `isocode.min.js`. For example, if you are launching Dothraki News, name the file `mis.min.js`.
```
Comment (RafaelDavisH) [inline docs/how-to-enable-new-languages.md:317]: I don't see this path `build/news-assets/dayjs/1.10.4/locale`. What I see is `build/news-assets/dayjs/days-1.9.1/locale`. Which path is the correct one? @scissorsneedfoodtoo

Comment (raisedadead) [reply docs/how-to-enable-new-languages.md:317]: image

Comment (raisedadead) [reply docs/how-to-enable-new-languages.md:317]: In any case, we should clear up version no. will change??

Comment (naomi-lgbt) [reply docs/how-to-enable-new-languages.md:317]:
```suggestion
In the [`build/news-assets/dayjs/<version>/locale`](https://github.com/freeCodeCamp/cdn/tree/main/build/news-assets/dayjs/1.10.4/locale) directory, create a new file and name it `isocode.min.js`. For example, if you are launching Dothraki News, name the file `mis.min.js`.

> [!NOTE]
> The version number will change as the dependencies are updated.
```

Comment (RafaelDavisH) [reply docs/how-to-enable-new-languages.md:317]: > In any case, we should clear up version no. will change??

Sorry, I deleted the comment before refreshing the page because with fresh eyes, I spotted the folder `1.10.4` . Also, I didn't see your comments. 

It's much clearer with the `<version>` placeholder 👍

File: docs/how-to-enable-new-languages.md [hunk 9]
```diff
@@ -281,3 +281,130 @@ Once these are in place, you should be able to run `npm run develop` to view you
...
+
+Then, visit [this page on cdnjs](https://cdnjs.com/libraries/dayjs/1.10.4) with all available dayjs v1.10.4 files, find the `https://cdnjs.cloudflare.com/ajax/libs/dayjs/1.10.4/locale/isocode.min.js` link for the new language, and open it in a new tab.
+
+Copy the dayjs code from the new tab into the new file you created. The code will be similar to this, but minified:
```
Comment (RafaelDavisH) [inline docs/how-to-enable-new-languages.md:321]: I am sure we are talking about the `dayjs.min.js` file here. I believe it would be better to add the complete file name instead of `daysjs`. WYA'T?

File: docs/how-to-enable-new-languages.md (hunk 7 above)
Comment (scissorsneedfoodtoo) [reply docs/how-to-enable-new-languages.md]: Thanks for catching that @Sembauke 👍 I must have added the incorrect spelling of indices to my system's dictionary at some point.

File: docs/how-to-enable-new-languages.md (hunk 8 above)
Comment (scissorsneedfoodtoo) [reply docs/how-to-enable-new-languages.md]: Sure, I'll see what I can do about removing the version number from other parts of the doc in case we change this in the future.

File: docs/how-to-enable-new-languages.md (hunk 9 above)
Comment (scissorsneedfoodtoo) [reply docs/how-to-enable-new-languages.md]: Hey @RafaelDavisH, sorry again if this was unclear.

I left a comment and a link to the file we need to add in your PR to the CDN repo: https://github.com/freeCodeCamp/cdn/pull/163#issuecomment-1280433546

Also, I tried to rewrite the Day.js bits so it's more clear that the only thing we need to add is a Day.js locale file for the new language.

File: docs/how-to-enable-new-languages.md [hunk 10]
```diff
@@ -281,3 +281,138 @@ Once these are in place, you should be able to run `npm run develop` to view you
...
+```
+
+Then commit and push your branch directly to the News repo.
+
+> [!NOTE]
+> You need to be on one of the teams with access to the News repo to push branches directly to News. Currently, only the dev, i18n, and staff teams are allowed to do this.
+
+Finally, open a PR for review.
+
+Once both your PRs to the CDN and News repo are merged, staff will create a couple more PRs to enable the new language and launch the site.
```
Comment (ojeytonwilliams) [inline docs/how-to-enable-new-languages.md:418]: Is the intention to merge these PRs whenever they've been through review or wait until you're ready to deploy?
```suggestion
Once both your PRs to the CDN and News repo have been approved, they can be merged. Deployment will be handled subsequently by the staff.
```
Given that deployment is out of scope, I don't think we need to give details.

File: docs/how-to-enable-new-languages.md [hunk 11]
```diff
@@ -281,3 +281,138 @@ Once these are in place, you should be able to run `npm run develop` to view you
...
+  english: 'news',
+  dothraki: 'news-mis'
+};
+```
+
+### Add the i18next JSON Files for the New Language
+
+Next, go to the `config/i18n/locales` directory, create a new folder, and give it the name of the new language you're adding. For example, if you're launching Dothraki News, create a new folder named `dothraki`.
+
+Then go to the `english` directory, copy the JSON files, and paste them into your new folder.
```
Comment (ojeytonwilliams) [inline docs/how-to-enable-new-languages.md:401]:
```suggestion
Then copy the JSON files from the `english` directory to your new folder.
```
Saves a few characters.

**Review by ojeytonwilliams — DISMISSED** [review]

File: docs/how-to-enable-new-languages.md (hunk 9 above)
Comment (RafaelDavisH) [reply docs/how-to-enable-new-languages.md]: Sounds good @scissorsneedfoodtoo. I made the change to days.min.js file and push it to the PR #163

File: docs/how-to-enable-new-languages.md (hunk 10 above)
Comment (scissorsneedfoodtoo) [reply docs/how-to-enable-new-languages.md:418]: Sure, this suggestion looks good. The main thing I want to get across is that there are a few more steps before News can be launched for the new language, and it's still clear with this wording.

File: docs/how-to-enable-new-languages.md (hunk 11 above)
Comment (scissorsneedfoodtoo) [reply docs/how-to-enable-new-languages.md:401]: Awesome. This looks good, too.

**Review by ojeytonwilliams — COMMENTED** [review]
The copy looks great. Nice work, @scissorsneedfoodtoo

I can't speak to the validity of the steps, though, so I'll leave approving it to people that do!

**Review by naomi-lgbt — APPROVED** [review]
Should be good to go.

**Review by raisedadead — APPROVED** [review]
Sneaking in a MERGE before someone nitpicks this.

**Comment by naomi-lgbt** [comment]
> Sneaking in a MERGE before someone nitpicks this.

You know me too well.

**Comment by scissorsneedfoodtoo** [comment]
Thanks everyone for all of your helpful suggestions. Not nitpicky at all :)

We can continue to improve this guide as other language leads go through the process of setting up a new language.
